setup/install.py

```diff
@@ -366,21 +366,25 @@ def add_options(self, parser):
 
     def pre_sub_commands(self, opts):
         tdir = self.j(self.d(self.SRC), 'translations')
-        clone_cmd = [
-            'git', 'clone', f'https://github.com/{self.TRANSLATIONS_REPO}.git', 'translations']
-        if opts.ephemeral:
-            if os.path.exists(tdir):
-                shutil.rmtree(tdir)
-
-            st = time.time()
-            clone_cmd.insert(2, '--depth=1')
-            subprocess.check_call(clone_cmd, cwd=self.d(self.SRC))
-            print('Downloaded translations in %d seconds' % int(time.time() - st))
+        env = os.getenv('TRANSLATIONS')
+        if env:
+            subprocess.check_call(['cp', '-r', '--no-preserve=mode,ownership', env, 'translations'])
         else:
```

Comment:
This all looks way too complicated, and I think it makes far more sense to allow `git pull` to fail here if the network is down or blocked.

It turns out that this is already possible, and I did it a long time ago: just copy the translations in on your own, and then run `for stage in build iso639 iso3166 translations gui resources cacerts recent_uas; do python setup.py $stage; done` instead of running bootstrap directly.

Reply:
I just keep all original logics in `else` block, the diff is misleading. I plan to submit commits related to the git pull logic here later.